src/query/runner/ipa.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{
+    error::Error,
+    ff::{FieldType, Fp31, Fp32BitPrime, GaloisField, PrimeField, Serializable},
+    helpers::{query::IpaQueryConfig, ByteArrStream},
+    protocol::{
+        attribution::input::MCAggregateCreditOutputRow,
+        context::SemiHonestContext,
+        ipa::{ipa, IPAInputRow},
+        BreakdownKey, MatchKey,
+    },
+    query::ProtocolResult,
+    secret_sharing::replicated::semi_honest::AdditiveShare,
+};
+use futures_util::StreamExt;
+use std::future::Future;
+use typenum::Unsigned;
+
+pub struct Runner(pub IpaQueryConfig);
+
+impl Runner {
+    pub async fn run(
+        &self,
+        ctx: SemiHonestContext<'_>,
+        field: FieldType,
+        input: ByteArrStream,
+    ) -> Box<dyn ProtocolResult> {
+        match field {
+            FieldType::Fp31 => Box::new(
+                self.run_internal::<Fp31, MatchKey, BreakdownKey>(ctx, input)
+                    .await
+                    .expect("IPA query failed"),
+            ),
+            FieldType::Fp32BitPrime => Box::new(
+                self.run_internal::<Fp32BitPrime, MatchKey, BreakdownKey>(ctx, input)
+                    .await
+                    .expect("IPA query failed"),
+            ),
+        }
+    }
+
+    /// It is intentionally made not async because it does not capture `self`.
```

Comment:
Nit: This is an implementation note, not a documentation note.

```suggestion
    // This is intentionally made not async because it does not capture `self`.
```